docs/src/content/docs/core/with-plugins.mdx

```diff
@@ -0,0 +1,24 @@
+---
+title: withPlugins
+description: Allows to extend hooks with custom parameters. Only used for advanced use-cases.
+---
+
+`withPlugins` allows to extend hooks with custom parameters. Useful for advanced use-cases.
+
+```ts title="my-integration/index.ts" "withPlugins"
+import { defineIntegration, withPlugins } from "astro-integration-kit";
+import { hasVitePluginPlugin  } from "astro-integration-kit/plugins";
+
+export default defineIntegration({
+    name: "my-integration",
+    setup({ name }) {
+        return withPlugins({
+            name,
```

Comment:
Unfortunately it has to be passed explicitly to decouple it from `defineIntegration`. That could be renamed to `integrationName` but I thought it would be more convenient to reuse the same name as the setup args. I don't know what we can improve here, apart being super clear in the documentation